src/firefly/js/charts/chartTypes/HistogramTblView.jsx

```diff
@@ -0,0 +1,156 @@
+    if (!TblUtil.isFullyLoaded(tblId) || !chartData) {
+        return null;
+    }
+    if (!heightPx || !widthPx) { return (<div/>); }
+    const { isDataReady, data:histogramData, options:histogramParams} = ChartsCntlr.getChartDataElement(chartId);
+
+    if (isDataReady) {
+        var logs, reversed;
+        if (histogramParams) {
+            var logvals = '';
+            if (histogramParams.x.includes('log')) { logvals += 'x';}
+            if (histogramParams.y.includes('log')) { logvals += 'y';}
+            if (logvals.length>0) { logs = logvals;}
+
+            var rvals = '';
+            if (histogramParams.x.includes('flip')) { rvals += 'x';}
+            if (histogramParams.y.includes('flip')) { rvals += 'y';}
+            if (rvals.length>0) { reversed = rvals;}
+
+        }
+
+        return (
+            <Histogram data={histogramData}
+                       desc={histogramParams.columnOrExpr}
+                       binColor='#8c8c8c'
+                       height={heightPx}
+                       width={widthPx}
+                       logs={logs}
+                       reversed={reversed}
+            />
+        );
+    } else {
+        if (histogramParams) {
+            return <div style={{position: 'relative', width: '100%', height: '100%'}}><div className='loading-mask'/></div>;
+        } else {
+            return 'Select Histogram Parameters';
+        }
+    }
+}
+
+function renderOptions(chartId, optionsKey) {
+    if (optionsKey === 'options') {
+        const {tblStatsData} = getChartProperties(chartId);
+        if (get(tblStatsData,'isColStatsReady')) {
+            const chartDataElement = ChartsCntlr.getChartDataElement(chartId);
+            const chartDataElementId = chartDataElement.id;
+            const formName = `chartOpt-${chartId}`;
+            return (
+                <HistogramOptions key={formName}
+                                  groupKey={formName}
+                                  colValStats={tblStatsData.colStats}
+                                  histogramParams={get(chartDataElement, 'options')}
+                                  defaultParams={get(chartDataElement, 'defaultOptions')}
+                                  onOptionsSelected={(options) => {
+                                        ChartsCntlr.dispatchChartOptionsReplace({chartId, chartDataElementId, newOptions: options});
+                                  }}
+                />
+            );
+        } else {
+            return (
+                <img style={{verticalAlign:'top', height: 16, padding: 10, float: 'left'}}
+                     title='Loading Options...'
+                     src={LOADING}
+                />);
+        }
+    } else if (optionsKey === 'filters') {
+        const {tableModel} = getChartProperties(chartId);
+        return (
+            <FilterEditorWrapper tableModel={tableModel}/>
+        );
+    }
+}
+
+function renderToolbar(chartId, expandable, expandedMode, toggleOptions) {
```

Comment:
I did make Chart, Toolbar, and Options functional components first, but then I wanted to switch to non-functional component for one chart type, and realized I can not pass it as a function anymore. May be it's not a big deal, and the recipe would be wrapping it into a functional component.